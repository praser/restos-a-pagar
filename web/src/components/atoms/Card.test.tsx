import React from 'react';

import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import {
  Card,
  CardHeader,
  CardBody,
  CardSmallText,
} from 'components/atoms/Card';

describe('Card atom component', () => {
  it('is expected to render the component correctly', () => {
    const card = renderer.create(<Card />).toJSON();
    expect(card).toMatchSnapshot();
  });

  it('is expected to have the right width if specified', () => {
    const { container } = render(<Card width="80px" />);
    expect(container.firstChild).toHaveStyle('width: 80px');
  });
});

describe('CardHeader atom component', () => {
  it('is expected to render the component correctly', () => {
    const cardHeader = renderer.create(<CardHeader />).toJSON();
    expect(cardHeader).toMatchSnapshot();
  });
});

describe('CardBody atom component', () => {
  it('is expected to render the component correctly', () => {
    const cardBody = renderer.create(<CardBody />).toJSON();
    expect(cardBody).toMatchSnapshot();
  });
});

describe('CardSmallText atom component', () => {
  it('is expected to render the component correctly', () => {
    const cardSmallText = renderer.create(<CardSmallText />).toJSON();
    expect(cardSmallText).toMatchSnapshot();
  });
});
