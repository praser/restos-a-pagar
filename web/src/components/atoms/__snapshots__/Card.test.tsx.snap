// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card atom component is expected to render the component correctly 1`] = `
<div
  className="sc-bdfBwQ gXFlcI"
/>
`;

exports[`CardBody atom component is expected to render the component correctly 1`] = `
<div
  className="sc-dlfnbm enEHeW"
/>
`;

exports[`CardHeader atom component is expected to render the component correctly 1`] = `
<div
  className="sc-gsTCUz biQnRI"
/>
`;

exports[`CardSmallText atom component is expected to render the component correctly 1`] = `
<p
  className="sc-hKgILt bcOtjz"
/>
`;
